client/pwa/src/db.ts

```diff
@@ -57,9 +91,51 @@ export class MDNOfflineDB extends Dexie {
       watched: "url, title, path",
       notifications: "id, title, text, url, created, read, starred",
     });
+    this.version(2).stores({
+      contentStatusHistory: "++id",
+    });
   }
 }
 
 const offlineDb = new MDNOfflineDB();
 
-export { offlineDb };
+async function getContentStatus(): Promise<ContentStatus> {
+  const current = await offlineDb.contentStatusHistory.toCollection().last();
+
+  return (
+    current || {
+      phase: ContentStatusPhase.INITIAL,
+      local: null,
+      remote: null,
+      progress: null,
+      timestamp: new Date(),
+    }
+  );
+}
+
+async function patchContentStatus(
```

Comment:
That's only used to migrate? Could we potentially use as message and do this in the service worker? Then we don't need this duplication and the client doesn't write to the DB.

Reply:
👍

Reply:
FYI This is tricky: If we use a message to migrate, but the service-worker is _outdated_ and doesn't know about that type of message yet, we will just lose our local status, and the user will have to download the whole content again. 🤔